### DISCUSSION TITLE
Roundtrip (Import/Export)

### DISCUSSION THREAD

**Discussion by michael-wynne-wsu** [MukurtuCMS/Mukurtu-CMS#5] [Mukurtu 4 Core Features for Launch]
Roundtrip is....

**Comment by michael-wynne-wsu** [comment]
User submitted:

Include import/export tools for taxonomies.

**Comment by MonicaGalassi** [comment]
@michael-wynne-wsu not sure which updates will be done for the roundtrip as I haven't used it much myself. However, I had different people asking me about easier ways to ingest content in Mukurtu that didn't require an excel spreadsheet. Drag and drop would be the easiest way if that's possible at this stage or in the future?

**Comment by michael-wynne-wsu** [comment]
Drag and drop of what exactly? Where does the metadata come from then? 

Batch import will likely always require some kind of structured metadata (and a spreadsheet is probably easier for most than an XML file, for example).

This is a bit too open ended to act on, but we can talk through what you mean...

**Comment by jpringle1980** [comment]
This is a known issue with our specific institutional needs, but hopefully the Roundtrip bulk import function will be improved to allow for bulk imports of much larger data sets/spreadsheets. We anticipate bringing on new content creators who may have legacy data that will require some massaging into the import template, but the bigger issue for us has been a slow import of only a few records at a time (seemingly a maximum of 10 per spreadsheet, itself taking 10 minutes or so) before Roundtrip stops working. Ideally we could import a single spreadsheet containing dozens, if not hundreds, of records at once.

**Reply by michael-wynne-wsu** [reply]
Thanks Jonathan,

The root of that issue is the Drupal 7 feeds module (and frustratingly it did work at scale in the past, but support for that module by the developers has tapered off). 

Import/export in Mukurtu 4 (on Drupal 9 or 10, tbd) will not be using the same underlying tools so we don't anticipate seeing that same issue develop. We will certainly be testing these tools at scale as they are refined.

Is there anything else you'd like to see fundamentally different about the import/export tools beyond general stability improvements?

**Reply by jpringle1980** [reply]
Thanks, Michael! I know you're all thinking about it and I appreciate the response. No additional thoughts about this; allowing for more expeditious/stable imports would be hugely helpful for us - and others as well I imagine.

**Reply by michael-wynne-wsu** [reply]
Oh, us as well, it's a big part of our project workflows. Thanks for chiming in!
